src/test/coll.rs

```diff
@@ -466,3 +466,109 @@ async fn empty_insert() {
         e => panic!("expected argument error, got {:?}", e),
     };
 }
+
+fn get_allow_disk_use(event_client: &EventClient) -> bool {
+    event_client
+        .command_events
+        .read()
+        .unwrap()
+        .iter()
+        .any(|event| match event {
+            CommandEvent::CommandStartedEvent(CommandStartedEvent {
+                command_name,
+                command,
+                ..
+            }) => {
+                if command_name == "find" {
+                    match command.get_bool("allowDiskUse") {
+                        Ok(b) => b,
+                        _ => false,
+                    }
+                } else {
+                    false
+                }
+            }
+            _ => false,
+        })
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn find_allow_disk_use() {
```

Comment:
I think these tests could be abstracted to share most of their code. I'm thinking that you could write a function with the signature `fn allow_disk_use_test(options: FindOptions, expected_value: Option<bool>)` that would run the test with the given options and assert that there's a find command with the option as specified.